src/components/books/bookform.js

```diff
@@ -1,15 +1,51 @@
-const BookForm = () => (
-  <section>
-    <h2 className="form-title"> ADD NEW BOOK</h2>
-    <form className="add-form">
-      <input className="input title-input" placeholder="Book title" required="" value="" />
-      <select className="input category-input">
-        <option value="Action">Action</option>
-        <option value="Science Fiction">Science Fiction</option>
-        <option value="Economy">Economy</option>
-      </select>
-      <button className="primary-button-big" type="submit">ADD BOOK</button>
-    </form>
-  </section>
-);
+import { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { addbook } from '../../redux/books/booksSlice';
+
+const BookForm = () => {
+  const [title, setTitle] = useState('');
+  const [author, setAuthor] = useState('');
+  const [category, setCategory] = useState('');
+  const dispatch = useDispatch();
+
+  const onAuthor = (e) => {
+    setAuthor(() => e.target.value);
+  };
+  const onTitle = (e) => {
+    setTitle(() => e.target.value);
+  };
+
+  const onCategory = (e) => {
+    setCategory(() => e.target.value);
+  };
+
+  const onaddbook = (e) => {
+    e.preventDefault();
+    if (title.trim() !== '' && author.trim() !== '' && category.trim() !== '') {
+      const book = {
+        title,
+        author,
+        category,
+      };
+      dispatch(addbook(book));
+    }
+  };
```

Comment:
## Recommendation Changes 

- [ ] It would be nice if you could clear the input fields as soon as a book is added so the user doesn't have to do that manually 👍🏽